src/mod/dns.mod/coredns.c

```diff
@@ -1201,6 +1202,109 @@ static void dns_lookup(sockname_t *addr)
   sendrequest(rp, T_PTR);
 }
 
+/* Read /etc/hosts. fgets() is used instead of mmap() because we prefer
+ * maintainability over optimized code, even though we read /etc/hosts for every
+ * lookup and /etc/hosts can be big, for example if used for stopping banner
+ * ads */
```

Comment:
The real optimization in that case would be caching file content based on mtime, not mmap

Reply:
yes, and if you what me to implement that, let me know.